packages/dd-trace/src/gateway/engine/runner.js

```diff
@@ -0,0 +1,40 @@
+'use strict'
+
+const als = require('../als')
+
+let lock = false // lock to prevent recursive calls to runSubscriptions
```

Comment:
How can this happen? Any other way to prevent that?

Reply:
It's a prevention measure we had back in Sqreen. I kept it because it's better to have it than debugging an infinite loop in a client's app.